src/misc/mmiocc.cc

```diff
@@ -0,0 +1,617 @@
+/* Copyright 2017-2018. Damien Nguyen.
+ * Copyright 2017-2018. Francesco Santini
+ * All rights reserved. Use of this source code is governed by
+ * a BSD-style license which can be found in the LICENSE file.
+ *
+ * Authors:
+ * 2017-2018 Nguyen Damien <damien.nguyen@alumni.epfl.ch>
```

Comment:
I would find it a bit more useful and fitting when this were written C, but ok.
I assume you want to be able to wrap existing memory buffers. Otherwise it would have been far simpler to simply named shared memory...

Reply:
Ok, I cherry-picked this to master with minor changes.